main.tf

```diff
@@ -16,17 +39,88 @@ resource "sumologic_monitor" "tf_monitor_1" {
       }
   }
 
-  dynamic "triggers" {
-    for_each = var.triggers
-    content {
-            threshold_type = triggers.value.threshold_type
-            threshold = triggers.value.threshold
-            time_range = triggers.value.time_range
-            occurrence_type = triggers.value.occurrence_type
-            trigger_source = triggers.value.trigger_source
-            trigger_type = triggers.value.trigger_type
-            detection_method = triggers.value.detection_method
-          }
+ trigger_conditions {
+    dynamic "logs_static_condition" {
+            for_each = toset(var.monitor_monitor_type == "Logs" ? ["1"] : [])
+            content {
+              dynamic "critical" {
+                for_each = local.hasLogsCriticalAlert ? ["1"] : []
+                content {
+                      time_range = local.LogsCriticalAlert[0].time_range
+                      alert {
+                        threshold      = local.LogsCriticalAlert[0].threshold
+                        threshold_type = local.LogsCriticalAlert[0].threshold_type
+                      }
+                      resolution {
+                        threshold      = local.LogsResolveCriticalAlert[0].threshold
+                        threshold_type = local.LogsResolveCriticalAlert[0].threshold_type
+                      }
+                    }
+              }
+              dynamic "warning" {
+                for_each = local.hasLogsWarningAlert ? ["1"] : []
+                content {
+                  time_range = local.LogsWarningAlert[0].time_range
+                  alert {
+                    threshold      = local.LogsWarningAlert[0].threshold
+                    threshold_type = local.LogsWarningAlert[0].threshold_type
+                  }
+                  resolution {
+                    threshold      = local.LogsResolveWarningAlert[0].threshold
+                    threshold_type = local.LogsResolveWarningAlert[0].threshold_type
+                  }
+                }
+              }
+            }
+    }
+    dynamic "logs_missing_data_condition" {
+      for_each = local.hasLogsMissingData ? ["1"] : []
+      content {
+        time_range = local.LogsMissingData[0].time_range
+      }
+    }
+    dynamic "metrics_static_condition" {
+            for_each = toset(var.monitor_monitor_type == "Metrics" ? ["1"] : [])
+            content {
+              dynamic "critical" {
+                for_each = local.hasMetricsCriticalAlert ? ["1"] : []
+                content {
+                      time_range = local.MetricsCriticalAlert[0].time_range
+                      occurrence_type = local.MetricsCriticalAlert[0].occurrence_type
```

Comment:
Are we not using any OOTB monitor with `occurrence_type` for `alert` as `Always` and for `resolution` as `AtLeastOnce`?

Reply:
In AWS O and k8s solution we are not using different occurrence_type for Alert and Resolution. Also as discussed with you this is rare corner case so we can pick it up in next iteration. Created a JIRA for the same https://jira.kumoroku.com/jira/browse/SUMO-206392